docs/docs/core/upload/backend/providers/intro.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Introduction
+slug: /upload
+tags:
+  - upload
+---
+
+# Provider
+
+Extends the upload plugin, to connect to different external services or applications, such as Amazon S3 buckets, Cloudinary, etc.
+
+**In the case of the Upload plugin, the provider should be able to upload files to a remote server and delete them.**
+
+# Using a provider
+
+To use a provider, you need to install it and configure it in the `./config/plugins.js` file.
+
+More info about installing providers on [strapi docs](https://docs.strapi.io/developer-docs/latest/development/providers.html#installing-providers).
+
+# Provider development
+
+To create a provider, you need to create a package that exports a function that returns an object with the following methods:
+
+- `isPrivate()` (optional) - Returns a boolean indicating if the provider is private or not. If it is, the `getSignedUrl` method will be used to get the URL of the file. (default: `false`)
+- `getSignedUrl(file)`. (optional) - Returns a signed URL to access the file if it requires authentication
```

Comment:
Hi, driving by with a quick question. According to this, it seems like this will be compatible with other providers (e.g. [this one](https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage) from the community for Google Cloud Storage). 

Once this PR is merged, would enabling access to private storage be as simple as adding these two new methods to the existing provider?

Reply:
Hello :) Yes! this PR will make possible adapting other providers (like GCS) by using the `isPrivate()` and `getSignedUrl(file)` methods.

We will create an experimental release this next week. If anyone is interested on adapting one of the GCS providers and test if this works it would be great to test this feature too!

Reply:
Awesome, thank you! I'm definitely interested in testing out the experimental release :D

Reply:
have you added the domain of your bucket in the middlewares?

example:
	```
	./config/middlewares.js

module.exports = [
  // ...
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            'yourBucketName.s3.yourRegion.amazonaws.com',
          ],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            'yourBucketName.s3.yourRegion.amazonaws.com',
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
  // ...
]
```

Reply:
and thank you for actually testing this on the release branch , this is very helpfull.

Reply:
This is what I currently have - would including `storage.googleapis.com` not be sufficient?
![Screenshot 2023-03-15 at 12 15 33 PM](https://user-images.githubusercontent.com/6666220/225372933-6b0cdccf-e66d-4d18-8674-86d03320d2a0.png)

Reply:
can you share the `host` part of one of the urls that the api returns? or just the url if possible

Reply:
for example for a s3 bucket I have

`<bucket-name>.s3.eu-west-3.amazonaws.com`

Reply:
Sure, the url looks like: https://storage.googleapis.com/[bucket name]/thumbnail_cat2.jpg?X-Goog-Algorithm=...(signature info)

Reply:
then storage.googleapis.com should be enough for your case. 

So, if you open that image directly from your browser it works. We could say it is being signed correctly.

It is on the frontend side that it's not even making the request to load the image right? It really sounds a problem from the middlewares config, but we discarded that.

Does the GCP work for you with a public configuration?

Reply:
ah, once you updated the strapi version to use the nightly version, have you built your project?

Reply:
re: public configuration: i haven't tried it myself, but assumed it would from reading others' experiences.

Hmm, I'm relatively certain I've rebuilt the project, but I can try again as a sanity check.

Reply:
Rebuilt the project and seeing the same behavior.

re: public config: Sure, I can give that a try. It still seems odd to me that no request is being sent to fetch the image though...

Reply:
yeah , that only should be the case when it's blocked by the strapi::security middleware

Reply:
is there another error on the browser console?

Reply:
hm... no, not seeing any errors there.

Reply:
I would first see if it happens for a public config then, it will point us if this is a config problem or there is smething wrong with this branch.

Reply:
sounds good, will circle back here once i confirm what happens with a public config. thanks!

Reply:
thank you for testing this !

Reply:
When I have `isPrivate()` in the provider return false, I can see that requests are made to fetch the image that fail with a 403 error (as expected). Since as you mentioned, we've ruled out the security middleware, this leads me to think that something may be wrong in this branch. What do you think?

And sure! Thank you for working on this :)

Reply:
I will try to replicate your setup this week and see what it is 🙂

Reply:
Thanks so much! Would be happy to clarify anything further and/or jump on a call if you'd like.

Reply:
Hello :) I have managed to test this actually, your code was actually super useful!

Everything seems to work ok on my setup :/  I am able to authenticate images and display them in the Media Library.

I find weird that only when the image is authenticated your browser doesn't make the request, could it be that something else is blocking it?

We could jump in a call next week monday if you are available :) 
Also, I would use this opportunity to open a pr in https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage to add this functionality.

Reply:
Interesting...it does seem weird that on my end, the browser doesn't make the request only when the image needs auth.

Sure, I'd be more than happy to jump on a call next Monday! Would sometime between 9am-1:30pm ET work?

And yes, also happy to open a PR in the GCS provider repo once I confirm the functionality on my end.

Reply:
Actually, Monday may not be a good time - maybe we can discuss further on Discord? Is your username there Marc-André?

Reply:
@jihyukbae  my discord user: MarcRoig#5524 

feel free to ping me whenever you are available :)

Reply:
@jihyukbae I created the PR in the meantime https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage/pull/165


Feel free to add any comment in there!! Let's see if we can find more people to test this